lib/src/map/options/map_options.dart

```diff
@@ -60,14 +31,20 @@ class MapOptions {
   final Color backgroundColor;
 
   final TapCallback? onTap;
-  final TapCallback? onSecondaryTap;
   final LongPressCallback? onLongPress;
-  final PointerDownCallback? onPointerDown;
-  final PointerUpCallback? onPointerUp;
-  final PointerCancelCallback? onPointerCancel;
-  final PointerHoverCallback? onPointerHover;
+  final TapCallback? onSecondaryTap;
+  final LongPressCallback? onSecondaryLongPress;
+  final TapCallback? onTertiaryTap;
+  final LongPressCallback? onTertiaryLongPress;
+
+  final void Function(PointerDownEvent event, LatLng point)? onPointerDown;
+  final void Function(PointerUpEvent event, LatLng point)? onPointerUp;
+  final void Function(PointerCancelEvent event, LatLng point)? onPointerCancel;
+  final void Function(PointerHoverEvent event, LatLng point)? onPointerHover;
+
   final PositionCallback? onPositionChanged;
-  final MapEventCallback? onMapEvent;
+
+  final void Function(MapEvent event)? onMapEvent;
```

Comment:
As mentioned above, are we sure loads of callbacks is the best way of doing things here?

Reply:
merging discussion with https://github.com/fleaflet/flutter_map/pull/1733#discussion_r1428964941

Reply:
discussion merged with https://github.com/fleaflet/flutter_map/pull/1733#discussion_r1428960703

Reply:
Closing this conversation since the discussion it got merged with is resolved.